[DataGridPremium] Pivoting (#9877)
mui/mui-x: docs/data/data-grid/overlays/overlays.md

{{"demo": "NoResultsOverlayCustom.js", "bg": "inline"}}

## Empty pivot overlay

The empty pivot overlay is displayed when pivot mode is enabled but fields have been added to the rows section.

{{"demo": "EmptyPivotOverlay.js", "bg": "inline"}}

### Custom component

> If you want to customize the empty pivot overlay, a component can be passed to the `emptyPivotOverlay` slot and rendered in place.

Reviewer: ```suggestion
To customize the empty pivot overlay, pass a custom component to the `emptyPivotOverlay` slot.
```
